### ISSUE TITLE
Interactive doc is not working with recursive urls

### ISSUE THREAD

**Issue by nanuxbe** [encode/django-rest-framework#4965]
## Steps to reproduce

Register 2 ViewSets with urls `todo/lists` and `todo/tasks` and a third unrelated url (likes `users`).
CoreAPI will then produce a document like this
```
<Document "http://localhost:8000/api/v1/docs/">
    todo: {
        lists: {
            list([page])
            create(name)
            read(id)
            update(id, name)
            partial_update(id, [name])
            delete(id)
        }
        tasks: {
            list([page], [done], [lst_id])
            create(lst, description, [done])
            read(id)
            update(id, lst, description, [done])
            partial_update(id, [lst], [description], [done])
            delete(id)
        }
    }
    users: {
        list([page], [first_name], [last_name])
        create(username, [email], [first_name], [last_name])
        read(id)
        update(id, username, [email], [first_name], [last_name])
        partial_update(id, [username], [email], [first_name], [last_name])
        delete(id)
    }
```

## Expected behavior

**In the menu:** see `todo` unfold and display `todo/lists` and `todo/tasks` which in turn will be able to unfold and display a list of leaf endpoints like for `users` or any "regular" viewset

**In the documentation section:** see the documentation for `todo/lists` and `todo/tasks` and be able to interact with it

## Actual behavior

**In the menu:** `todo` is visible but nothing happens when clicked

**In the documentation section:** the title `todo` is displayed but there is no documentation underneath it.

**Comment by myrubapa** [comment]
Hi!. I am apply your change, good job!. Can you some issue?

1. _recursive_menu.html put ul in inner li tag 
2. I can't distinguish sub menu from parent  menu give him recursive padding left 

Thanks

**Comment by carltongibson** [comment]
Closed by #5334
